Home_Work_20/Victory.cpp

```diff
@@ -0,0 +1,53 @@
+#include "Victory.h"
+#include "Munchkin.h"
+#include "ItemDeck.h"
+#include <vector>
+#include "DeckSelectionHelper.h"
+#include "Item.h"
+#include "Modifier.h"
+#include "ModifierDeck.h"
+
+void Victory_AdditionalLevels::apply(Munchkin* munchkin, ItemDeck* items, ModifierDeck* modifiers)
+{
+	munchkin->updateLevelBy(m_additionalLevels);
+}
+
+	
+void Victory_AdditionalItems::apply(Munchkin* munchkin, ItemDeck* items, ModifierDeck* modifiers)
+{
+	
+	std::vector<Item* > generetadItems;
+	
+
+	while (generetadItems.size() < m_additionalItems)
+	{
+		generetadItems.push_back(items->generateItem());
+		
+		
+	}
+	
+	while (generetadItems.size()!=0)
+	{
+		munchkin->addItem(generetadItems[0]);
+		generetadItems.erase(generetadItems.begin());
+	}
+}
+
+void Victory_AdditionalModifiers::apply(Munchkin* munchkin, ItemDeck* items, ModifierDeck* modifiers)
+{
+	std::vector<Modifier* > generetadModifier;
+	
+
+	while (generetadModifier.size() < m_additionalModifiers)
+	{
+		generetadModifier.push_back(modifiers->generateModifier());
+	}
```

Comment:
the condition works fine but it looks strange 😄 Also, .size() method is called every iteration, and if it's not simple `return m_size` which we cannot guarantee then it may be even a little bit longer to execute.
`(for int i = 0; i < m_number; i++)` would be much cleaner